cpp/src/parquet/encoding.cc

```diff
@@ -2723,6 +2730,7 @@ class DeltaLengthByteArrayDecoder : public DecoderImpl,
     // Decode up to `max_values` strings into an internal buffer
     // and reference them into `buffer`.
     max_values = std::min(max_values, num_valid_values_);
+    DCHECK_GE(max_values, 0);
```

Comment:
should this be an exception?

Reply:
That might be safer, but as-is we should never trip that. I added this in case someone refactors the class and adds another code path where `num_valid_values_` might not be initialized. Do you still think it's worthwhile?